hazelcast/src/main/java/com/hazelcast/spi/impl/NodeEngineImpl.java

```diff
@@ -168,6 +174,18 @@ public NodeEngineImpl(Node node) {
         }
     }
 
+    private ConcurrencyDetection newConcurrencyDetection() {
+        HazelcastProperties properties = node.getProperties();
+        boolean writeThrough = properties.getBoolean(GroupProperty.IO_WRITE_THROUGH_ENABLED);
+        boolean backPressureEnabled = properties.getBoolean(BACKPRESSURE_ENABLED);
+
+        if (writeThrough || backPressureEnabled) {
```

Comment:
What's the purpose of enabling concurrency detection when backpressure is enabled if we only use concurrency detection in conjuction with write-through on member side (in `NioOutboundPipeline#schedule`)?

Reply:
Ah, found it, ignore.

Reply:
The spreading of properties is messy.